src/managed/Microsoft.NET.HostModel/AppHost/MachOUtils.cs

```diff
@@ -0,0 +1,300 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.IO;
+using System.IO.MemoryMappedFiles;
+using System.Runtime.CompilerServices;
+using System.Text;
+
+namespace Microsoft.NET.HostModel.AppHost
+{
+    public static class MacOUtils
```

Comment:
Does this need to be public? Is it for tests? Consider internals visible to tests.